contracts/VerifierRegistry.sol

```diff
@@ -31,19 +33,22 @@ contract VerifierRegistry is Ownable {
 
   mapping(address => Verifier) public verifiers;
 
+  mapping(uint256 => uint256) public totalBalancePerShard;
```

Comment:
`totalBalancePerShard` seems a bit redundant, also `balance` should be plural.

How about just: `balances`.

Reply:
is `balancesPerShard` would be ok? `balances` are usually uses as mapping for addresses, also we have `veriefiersPerShard` variable, so I follow this convention.